program-runtime/src/executor_cache.rs

```diff
@@ -14,80 +14,69 @@ use {
     },
 };
 
-/// Relation between a TransactionExecutorCacheEntry and its matching BankExecutorCacheEntry
-#[repr(u8)]
-#[derive(Clone, Copy, PartialEq, Eq, Debug)]
-pub enum TxBankExecutorCacheDiff {
-    /// The TransactionExecutorCacheEntry did not change and is the same as the BankExecutorCacheEntry.
-    None,
-    /// The TransactionExecutorCacheEntry was inserted, no matching BankExecutorCacheEntry exists, so it needs to be inserted.
-    Inserted,
-    /// The TransactionExecutorCacheEntry was replaced, the matching BankExecutorCacheEntry needs to be updated.
-    Updated,
-}
-
-/// An entry of the TransactionExecutorCache
-#[derive(Debug)]
-pub struct TransactionExecutorCacheEntry {
-    executor: Arc<dyn Executor>,
-    difference: TxBankExecutorCacheDiff,
-}
-
 /// A subset of the BankExecutorCache containing only the executors relevant to one transaction
 ///
 /// The BankExecutorCache can not be updated directly as transaction batches are
 /// processed in parallel, which would cause a race condition.
 #[derive(Default, Debug)]
 pub struct TransactionExecutorCache {
-    pub executors: HashMap<Pubkey, TransactionExecutorCacheEntry>,
+    /// Executors which are visible to the transaction
+    pub executors: HashMap<Pubkey, Option<Arc<dyn Executor>>>,
+    /// Executors of programs which were deploymed during the transaction
+    pub deployments: HashMap<Pubkey, Arc<dyn Executor>>,
+    /// Executors which were loaded after the transaction was loaded (because they were missing in the cache)
+    pub loaded: HashMap<Pubkey, Arc<dyn Executor>>,
```

Comment:
No only `executors` is used for visibility, `loaded` is just for updating the banks cache later.

Reply:
Ah, sorry, I didn't word my original comment properly. I meant to ask if executors *added* to `loaded` are meant to be visible immediately since they are also added to `executors` at same time in `set()` function

Reply:
Only executor which were not loaded before the transaction and were loaded during the transaction, but were **not re-/deployed within the transaction** go into `loaded`.